projects/plugins/jetpack/modules/masterbar/inline-help/class-inline-help.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Inline Help
+ *
+ * Handles providing a LiveChat icon within WPAdmin until such time
+ * as the full live chat experience can be run in a non-Calypso environment.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+use Automattic\Jetpack\Redirect;
+use Automattic\Jetpack\Status;
+
+/**
+ * Class Inline_Help.
+ */
+class Inline_Help {
+
+
+	public function __construct() {
+		$this->register_actions();
+	}
+
+	public function register_actions() {
+		add_action( 'admin_footer', array( $this, 'add_fab_icon' ) );
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'add_fab_styles' ) );
+	}
+
+	public function add_fab_icon( $data ) {
+
+		if ( wp_doing_ajax() ) {
+			return;
+		}
+
+		$is_gutenframe = 0 === strpos( $_SERVER['REQUEST_URI'], 'frame-nonce' );
+
+		if ( $is_gutenframe ) {
+			return;
+		}
```

Comment:
We can do a more specific target by looking for the `frame-nonce` GET param instead of checking that `framce-nonce` is present on the URL.

Also, the presence of the `frame-nonce` GET param does not necessarily indicates a Gutenframe context, since it can be used in other places (e.g. [post preview](https://github.com/Automattic/jetpack/blob/d03c9e834b52ba5a489ad2fd991f11f72f119f7e/projects/plugins/jetpack/class.frame-nonce-preview.php#L44)). Due to that, I think it's better to don't use any reference to Gutenframe in the variables.

```suggestion
		if ( ! empty( $_GET['frame-nonce'] ) {
			return;
		}
```

Reply:
Good points.